Problem: using uninitialized variable (linux/clang)

```cpp
In file included from ./src/v1_decoder.hpp:33:
./src/decoder.hpp:50:46: error: field 'bufsize' is uninitialized when used here
      [-Werror,-Wuninitialized]
                buf((unsigned char*) malloc (bufsize) ),
                                             ^
```
See: https://travis-ci.org/libbitcoin/libbitcoin-server/jobs/65729336#L1111

osx/clang and linux/gcc are okay

--- a/src/decoder.hpp
+++ b/src/decoder.hpp
@@ -47,8 +47,8 @@ namespace zmq
     {
     public:
         c_single_allocator(size_t bufsize_):
-                buf((unsigned char*) malloc (bufsize) ),
-                bufsize(bufsize_)
+                bufsize(bufsize_),
+                buf((unsigned char*) malloc (bufsize))
         {
             alloc_assert (buf);
         }
@@ -74,8 +74,8 @@ namespace zmq
         }
 
     private:
-        unsigned char* buf;
         size_t bufsize;
+        unsigned char* buf;
 
         c_single_allocator( c_single_allocator const& );
         c_single_allocator& operator=(c_single_allocator const&);

